File: app/src/main/java/com/example/petrsumobile/news/News.java
package com.example.petrsumobile.news;

import java.util.Date;

public class News {

    private String title;
    private String description;
    private String link;
    private Date date;
    private String imageUrl;

    public News(String title, String description, Date date, String link, String imageUrl){
        this.title= title;
        this.description = description;
        this.link = link;
        this.date = date;
        this.imageUrl = imageUrl;
    }

    public String getTitle(){
        return title;
    }
    public String getDescription(){
        return description;
    }
    public String getLink(){
        return link;
    }
    public Date getDate(){
        return date;
    }
    public String getImageUrl(){
        return imageUrl;
    }

}

File: app/src/main/java/com/example/petrsumobile/news/News.kt
package com.example.petrsumobile.news

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class NewsList(
        @Expose
        @SerializedName("items")
        var newsList: List<News>
)

@Entity
data class News(
        @PrimaryKey(autoGenerate = true)
        var id: Int = 0,
        @Expose
        var title: String,
        @Expose
        var description: String,
        @SerializedName("pubDate")
        @Expose
        var date: String,
        @Expose
        var link: String,
        @Embedded
        @Expose
        var enclosure: Enclosure
) {
    data class Enclosure(
            @Expose
            @SerializedName("link")
            var urlImage: String,
    )
}

